framework/src/loops/MaxQpsThread.C

```diff
@@ -64,19 +64,41 @@ MaxQpsThread::operator()(const ConstElemRange & range)
     // We'll use one for element interiors, which calculates nothing
     std::unique_ptr<FEBase> fe(FEBase::build(dim, fe_type));
     fe->get_nothing();
+    // Alex: This seems wrong, e.g. like we'd want the quadrature order to be higher if the
+    // polynomial basis is higher, but our fe_type up above is first Lagrange, so it seems like the
+    // person who wrote this code didn't want to consider higher order bases even if if the user
+    // asked for higher order bases
+    fe->add_p_level_in_reinit(false);
```

Comment:
I think this is wrong.

Contrary to the comment, our `fe_type` here is irrelevant, isn't it?  `attachQRuleElem` later attaches the quadrature rule to `fe`, but doesn't actually inspect `fe` to decide what quadrature rule to attach.  And then, within `libMesh::FE` itself, we don't look at the FE type when constructing quadrature points or weights - if someone has a quadratic and a cubic variable then in general they want to evaluate both on the same set of points, whatever fixed rule the user code has determined that to be ... and if the user really does want two different sets of points, they can get that through two different quadrature rules.

But!  `libMesh::Quadrature` *does* have an `unsigned int p_level = 0` argument to `init()`, which is filled with each `elem->p_level()` and which ought to be used (except for e.g. nodal quadrature rules where we ignore order) to *un*-fix the rule, to elevate the order of the rule by twice that `p_level`, via internal use of `Quadrature::get_order()`.  And we'd want to do the same here, to get an accurate count of qps.

Reply:
Maybe it's because I haven't been sleeping much, but I'm not clear on what alternative code you're suggesting

Reply:
Code?  None.  I'd just leave in a comment about how we want to retain `add_p_level_in_p_init(true)` so that our qps count here matches the number of qps that will later be allocated when assembling.

Reply:
Although now that I think about it, there's got to also be some place where we can stick a friendly assertion that `qrule.size()` doesn't exceed what `MaxQpsThread::max()` concluded, so we can have a failure of that instead of an OOB index who-knows-where.

Reply:
I'm still confused. I had to add this here such that upon recovery/restart I don't try to reinit a Lagrange FE with a p order it can't handle. You want me to remove this code or keep it as is?

Reply:
Oh, hell, we're using `LAGRANGE` in `fe` here.  Can we just switch that to `HIERARCHIC`?  We're just doing a `get_nothing()` on it, so it's probably just an overzealous assertion that's whining about the fact that it can't calculate things you never asked it to calculate.

Reply:
Ok I think what I'm going to do now that we have removed the max shape functions is just remove the FE from this routine. What do you think?

Reply:
My comment was indeed wrong, I see that now. I forgot that this is *not* the code that determines the quadrature order we use in our calculations. Indeed we are only querying the see what the maximum number of qps is based on the quadrature rules we've already set

Reply:
Remove the FE, and manually pass `elem->p_level()` to `Quadrature::init`?  Yeah, that's much smarter than my `HIERARCHIC` idea.